lib/engine/game/g_1826/game.rb

```diff
@@ -64,68 +64,73 @@ class Game < Game::Base
                   },
                   {
                     name: 'TVG',
                     on: 'TVG',
                     train_limit: 2,
                     tiles: %i[yellow green brown blue gray],
                     operating_rounds: 3,
-                    status: ['can_buy_trains'],
                   }].freeze
 
-        TRAINS = [
-                    { name: '2H', distance: 2, price: 100, rusts_on: '6H', num: 8 },
-                    { name: '4H', distance: 4, price: 200, rusts_on: '10H', num: 7 },
-                    {
-                      name: '6H',
-                      distance: 6,
-                      price: 300,
-                      rusts_on: 'E',
-                      num: 6,
-                      events: [{ 'type' => 'can_buy_trains' }],
-                    },
-                    {
-                      name: '10H',
-                      distance: 10,
-                      price: 600,
-                      num: 5,
-                    },
-                    {
-                      name: 'E',
-                      # distance is equal to the number of E and TGV trains in play. The run is doubled until a TVG is purchased.
-                      distance: [{ 'nodes' => %w[city offboard], 'pay' => 99, 'visit' => 99, 'multiplier' => 2 },
-                                 { 'nodes' => ['town'], 'pay' => 0, 'visit' => 99 }],
-                      price: 800,
-                      num: 2,
-                      events: [{ 'type' => 'remove_abilities' }],
-                    },
-                    {
-                      name: 'TGV',
-                      distance: [{ 'nodes' => %w[city offboard], 'pay' => 3, 'visit' => 99, 'multiplier' => 2 },
-                                 { 'nodes' => ['town'], 'pay' => 0, 'visit' => 99 }],
-                      price: 1000,
-                      num: 20,
-                      discount: { '4' => 300, '5' => 300, '6' => 300 },
-                    },
-                  ].freeze
+        def game_trains
+          [
+            { name: '2H', distance: 2, price: 100, rusts_on: '6H', num: 8 },
+            { name: '4H', distance: 4, price: 200, rusts_on: '10H', num: 7 },
+            {
+              name: '6H',
+              distance: 6,
+              price: 300,
+              rusts_on: 'E',
+              num: 6,
+              events: [{ 'type' => 'can_buy_trains' }],
+            },
+            {
+              name: '10H',
+              distance: 10,
+              price: 600,
+              num: 5,
+            },
+            {
+              name: 'E',
+              # distance is equal to the number of E and TGV trains in play. The run is doubled until a TVG is purchased.
+              distance: [
+                {
+                  'nodes' => %w[city offboard],
+                  'pay' => @e_train_range,
+                  'visit' => @e_train_range,
+                  'multiplier' => @tgv_purchased ? 1 : 2,
```

Comment:
Have you tested this? I would have thought that these variables would be evaluated when the train object is first created, not when the train is being used.